frame/nfts/src/tests.rs

```diff
@@ -398,53 +426,57 @@ fn set_item_metadata_should_work() {
+		assert_ok!(Nfts::mint(RuntimeOrigin::signed(1), 0, 42, 1, default_item_config()));
 		// Cannot add metadata to unowned item
 		assert_noop!(
-			Nfts::set_metadata(RuntimeOrigin::signed(2), 0, 42, bvec![0u8; 20], false),
+			Nfts::set_metadata(RuntimeOrigin::signed(2), 0, 42, bvec![0u8; 20]),
 			Error::<Test>::NoPermission,
 		);
 
 		// Successfully add metadata and take deposit
-		assert_ok!(Nfts::set_metadata(RuntimeOrigin::signed(1), 0, 42, bvec![0u8; 20], false));
+		assert_ok!(Nfts::set_metadata(RuntimeOrigin::signed(1), 0, 42, bvec![0u8; 20]));
 		assert_eq!(Balances::free_balance(&1), 8);
 		assert!(ItemMetadataOf::<Test>::contains_key(0, 42));
 
 		// Force origin works, too.
-		assert_ok!(Nfts::set_metadata(RuntimeOrigin::root(), 0, 42, bvec![0u8; 18], false));
+		assert_ok!(Nfts::set_metadata(RuntimeOrigin::root(), 0, 42, bvec![0u8; 18]));
 
 		// Update deposit
-		assert_ok!(Nfts::set_metadata(RuntimeOrigin::signed(1), 0, 42, bvec![0u8; 15], false));
+		assert_ok!(Nfts::set_metadata(RuntimeOrigin::signed(1), 0, 42, bvec![0u8; 15]));
 		assert_eq!(Balances::free_balance(&1), 13);
-		assert_ok!(Nfts::set_metadata(RuntimeOrigin::signed(1), 0, 42, bvec![0u8; 25], false));
+		assert_ok!(Nfts::set_metadata(RuntimeOrigin::signed(1), 0, 42, bvec![0u8; 25]));
 		assert_eq!(Balances::free_balance(&1), 3);
 
 		// Cannot over-reserve
 		assert_noop!(
-			Nfts::set_metadata(RuntimeOrigin::signed(1), 0, 42, bvec![0u8; 40], false),
+			Nfts::set_metadata(RuntimeOrigin::signed(1), 0, 42, bvec![0u8; 40]),
 			BalancesError::<Test, _>::InsufficientBalance,
 		);
 
 		// Can't set or clear metadata once frozen
-		assert_ok!(Nfts::set_metadata(RuntimeOrigin::signed(1), 0, 42, bvec![0u8; 15], true));
+		assert_ok!(Nfts::set_metadata(RuntimeOrigin::signed(1), 0, 42, bvec![0u8; 15]));
+		assert_ok!(Nfts::lock_item(RuntimeOrigin::signed(1), 0, 42, true, false));
+		assert_noop!(
+			Nfts::set_metadata(RuntimeOrigin::signed(1), 0, 42, bvec![0u8; 15]),
+			Error::<Test, _>::LockedItemMetadata,
+		);
 		assert_noop!(
-			Nfts::set_metadata(RuntimeOrigin::signed(1), 0, 42, bvec![0u8; 15], false),
-			Error::<Test, _>::Frozen,
+			Nfts::clear_metadata(RuntimeOrigin::signed(1), 0, 42),
+			Error::<Test>::LockedItemMetadata,
 		);
-		assert_noop!(Nfts::clear_metadata(RuntimeOrigin::signed(1), 0, 42), Error::<Test>::Frozen);
 
 		// Clear Metadata
-		assert_ok!(Nfts::set_metadata(RuntimeOrigin::root(), 0, 42, bvec![0u8; 15], false));
+		assert_ok!(Nfts::set_metadata(RuntimeOrigin::root(), 0, 42, bvec![0u8; 15]));
 		assert_noop!(
 			Nfts::clear_metadata(RuntimeOrigin::signed(2), 0, 42),
-			Error::<Test>::NoPermission
+			Error::<Test>::NoPermission,
 		);
 		assert_noop!(
 			Nfts::clear_metadata(RuntimeOrigin::signed(1), 1, 42),
-			Error::<Test>::UnknownCollection
+			Error::<Test>::UnknownCollection,
 		);
-		assert_ok!(Nfts::clear_metadata(RuntimeOrigin::signed(1), 0, 42));
+		assert_ok!(Nfts::clear_metadata(RuntimeOrigin::root(), 0, 42));
```

Comment:
Why does this now have to be root?

Reply:
When the metadata is locked, only the root has rights to clear it. There is a check for that a couple of line above:
```
		assert_noop!(
			Nfts::clear_metadata(RuntimeOrigin::signed(1), 0, 42),
			Error::<Test>::LockedItemMetadata,
		);
```